databricks/sdk/mixins/sql.py

```diff
@@ -0,0 +1,231 @@
+        return self[idx]
+
+    def __getitem__(self, col):
+        # if columns are named `2 + 2`, for example
+        return self.__getattr__(col)
+
+    def __repr__(self):
+        return f"Row({', '.join(f'{k}={v}' for (k, v) in zip(self.__columns__, self))})"
+
+
+class StatementExecutionExt(StatementExecutionAPI):
+
+    def __init__(self, api_client):
+        super().__init__(api_client)
+        self.type_converters = {
+            ColumnInfoTypeName.ARRAY: json.loads,
+            # ColumnInfoTypeName.BINARY: not_supported(ColumnInfoTypeName.BINARY),
+            ColumnInfoTypeName.BOOLEAN: bool,
+            # ColumnInfoTypeName.BYTE: not_supported(ColumnInfoTypeName.BYTE),
+            ColumnInfoTypeName.CHAR: str,
+            # ColumnInfoTypeName.DATE: not_supported(ColumnInfoTypeName.DATE),
+            ColumnInfoTypeName.DOUBLE: float,
+            ColumnInfoTypeName.FLOAT: float,
+            ColumnInfoTypeName.INT: int,
+            # ColumnInfoTypeName.INTERVAL: not_supported(ColumnInfoTypeName.INTERVAL),
+            ColumnInfoTypeName.LONG: int,
+            ColumnInfoTypeName.MAP: json.loads,
+            ColumnInfoTypeName.NULL: lambda _: None,
+            ColumnInfoTypeName.SHORT: int,
+            ColumnInfoTypeName.STRING: str,
+            ColumnInfoTypeName.STRUCT: json.loads,
+            ColumnInfoTypeName.TIMESTAMP: self._parse_timestamp,
+            # ColumnInfoTypeName.USER_DEFINED_TYPE: not_supported(ColumnInfoTypeName.USER_DEFINED_TYPE),
+        }
+
+    @staticmethod
+    def _parse_timestamp(value: str):
+        # make it work with Python 3.7 to 3.10 as well
+        return datetime.datetime.fromisoformat(value.replace('Z', '+00:00'))
+
+    @staticmethod
+    def _raise_if_needed(status: StatementStatus):
+        if status.state not in [StatementState.FAILED, StatementState.CANCELED, StatementState.CLOSED]:
+            return
+        msg = status.state.value
+        if status.error is not None:
+            msg = f"{msg}: {status.error.error_code.value} {status.error.message}"
+        raise RuntimeError(msg)
+
+    def execute(self,
+                warehouse_id: str,
+                statement: str,
+                *,
+                byte_limit: Optional[int] = None,
+                catalog: Optional[str] = None,
+                schema: Optional[str] = None,
+                timeout: timedelta = timedelta(minutes=20),
+                ) -> ExecuteStatementResponse:
+        """(Experimental) Execute a SQL statement and block until results are ready,
+        including starting the warehouse if needed.
+
+        This is a high-level implementation that works with INLINE result disposition,
+        fetching records in JSON format. It can be considered as a quick way to run SQL
+        queries that return up to a dozen megabytes of results by just depending on
+        Databricks SDK for Python without having to pull hundreds of megabytes of
+        library dependencies of Apache Arrow.
```

Comment:
Can we leave out the `hundreds of megabytes of library dependencies of Apache Arrow`, the tone is not great and makes Arrow sound very negative. In several cases it's the preferred format.